web-ui/src/components/reviews/periods/DatePickerField.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { useRef, useEffect, useState } from 'react';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import './DatePickerField.css';
+
+export default function DatePickerField({ date, setDate, label, disabled, open}) {
+
+  const [cleared, setCleared] = useState(false);
+  const launchDatePickerRef = useRef(null);
+
+  useEffect(() => {
+    if (cleared) {
+      const timeout = setTimeout(() => {
+        setCleared(false);
+      }, 1500);
```

Comment:
This bit is interesting to me. What makes this timeout needed?

Reply:
This is basically a debounce related to field clearing.

Reply:
See if this also works. I suspect it will and will avoid the arbitrary timeout of 1500.

```js
        requestAnimationFrame(() => {
          setCleared(false);
        });
```

Reply:
Testing - looks like this is working.